### PR TITLE
Enable casting to pandas nullable dtypes

### PR DESCRIPTION
**Pull request by brandon-b-miller**
Fixes https://github.com/rapidsai/cudf/issues/8885

Defer to the closest corresponding NumPy dtype.

### CODE REVIEW

File: python/cudf/cudf/core/column/column.py [hunk 1]
```diff
@@ -877,6 +878,9 @@ def can_cast_safely(self, to_dtype: Dtype) -> bool:
         raise NotImplementedError()
 
     def astype(self, dtype: Dtype, **kwargs) -> ColumnBase:
+        if dtype in cudf_dtypes_to_pandas_dtypes.values():
```
Comment (galipremsagar) [inline python/cudf/cudf/core/column/column.py:881]:
```suggestion
        if dtype in pandas_dtypes_to_cudf_dtypes:
```

Could use an inverse map that already exists.

**Review by galipremsagar — CHANGES_REQUESTED** [review]

File: python/cudf/cudf/core/column/column.py [hunk 2]
```diff
@@ -67,6 +67,7 @@
 from cudf.utils.dtypes import (
     check_cast_unsupported_dtype,
     cudf_dtype_from_pa_type,
+    cudf_dtypes_to_pandas_dtypes,
```
Comment (galipremsagar) [inline python/cudf/cudf/core/column/column.py:70]:
```suggestion
    pandas_dtypes_to_cudf_dtypes,
```

File: python/cudf/cudf/utils/dtypes.py [hunk 3]
```diff
@@ -62,6 +62,8 @@
     np.dtype("int16"): pd.Int16Dtype(),
     np.dtype("int32"): pd.Int32Dtype(),
     np.dtype("int64"): pd.Int64Dtype(),
+    np.dtype("float32"): pd.Float32Dtype(),
```
Comment (galipremsagar) [inline python/cudf/cudf/utils/dtypes.py:65]: Will need to do this inverse addition in `pandas_dtypes_to_cudf_dtypes` too..

File: python/cudf/cudf/core/column/column.py (hunk 1 above)
Comment (brandon-b-miller) [reply python/cudf/cudf/core/column/column.py:881]: lol, I did not even see that 👍 🤦

File: python/cudf/cudf/utils/dtypes.py (hunk 3 above)
Comment (galipremsagar) [reply python/cudf/cudf/utils/dtypes.py:65]: Never mind, we don't need this as it's being handled below in line 97 with proper versioning.:
```python
if PANDAS_GE_120:
    cudf_dtypes_to_pandas_dtypes[np.dtype("float32")] = pd.Float32Dtype()
    cudf_dtypes_to_pandas_dtypes[np.dtype("float64")] = pd.Float64Dtype()
    pandas_dtypes_to_cudf_dtypes[pd.Float32Dtype()] = np.dtype("float32")
    pandas_dtypes_to_cudf_dtypes[pd.Float64Dtype()] = np.dtype("float64")
```

**Review by galipremsagar — APPROVED** [review]

**Comment by brandon-b-miller** [comment]
Couple of test failures here I am looking into.

File: python/cudf/cudf/tests/test_joining.py [hunk 4]
```diff
@@ -1529,7 +1529,7 @@ def test_categorical_typecast_inner_one_cat(dtype):
     data = np.array([1, 2, 3], dtype=dtype)
 
     left = make_categorical_dataframe(data)
-    right = left.astype(left["key"].dtype.categories)
+    right = left.astype(left["key"].dtype.categories.dtype)
```
Comment (brandon-b-miller) [inline python/cudf/cudf/tests/test_joining.py:1532]: These were passing by coincidence due to https://github.com/rapidsai/cudf/issues/8895

**Review by shwina — APPROVED** [review]

**Comment by shwina** [comment]
@gpucibot merge

**Pull request merged**